packages/ag-charts-community/src/chart/interaction/interactionManager.ts

```diff
@@ -147,12 +143,31 @@ export class InteractionManager extends BaseManager<
         }
     }
 
-    resume(pauseType: PauseType) {
-        this.pausers[pauseType]--;
+    // Wrapper to only broadcast events when the InteractionManager is a given state.
+    override addListener<T extends InteractionTypes>(
+        type: T,
+        handler: (event: InteractionEvent<T> & { type: T }) => void,
+        triggeringStates: InteractionState = InteractionState.Default
+    ) {
+        return super.addListener(type, (e) => {
+            if (this.state & triggeringStates) {
+                handler(e);
+            }
+        });
     }
 
-    pause(pauseType: PauseType) {
-        this.pausers[pauseType]++;
+    public pushState(state: InteractionState) {
+        this.stateQueue.add(state);
+        this.updateCurrentState();
+    }
+
+    public popState(state: InteractionState) {
+        this.stateQueue.delete(state);
+        this.updateCurrentState();
+    }
+
+    private updateCurrentState() {
+        this.state = Array.from(this.stateQueue).reduce((max, s) => Math.max(max, s), InteractionState.Default);
```

Comment:
```suggestion
        this.state = Math.max(...this.stateQueue, InteractionState.Default);
```
nit: I personally think it's easier to read without using `reduce`